From 25b5155f5dcd55aa29fda89780c5d42fc89b94e9 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:05 -0400
Subject: [virt] kvm/x86: add MSR_IA32_BNDCFGS to msrs_to_save

Message-id: <1406040016-3289-151-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86229
O-Subject: [RHEL7 PATCH v2 150/221] KVM: x86: add MSR_IA32_BNDCFGS to msrs_to_save
Bugzilla: 1076194 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Liu, Jinsong" <jinsong.liu@intel.com>

Add MSR_IA32_BNDCFGS to msrs_to_save, and corresponding logic
to kvm_get/set_msr().

Signed-off-by: Liu Jinsong <jinsong.liu@intel.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 0dd376e709975779cf43f368498c5c0eec843b02)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 00b5993..53c324f 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2484,6 +2484,9 @@ static int vmx_get_msr(struct kvm_vcpu *vcpu, u32 msr_index, u64 *pdata)
  case MSR_IA32_SYSENTER_ESP:
   data = vmcs_readl(GUEST_SYSENTER_ESP);
   break;
+ case MSR_IA32_BNDCFGS:
+  data = vmcs_read64(GUEST_BNDCFGS);
+  break;
  case MSR_IA32_FEATURE_CONTROL:
   if (!nested_vmx_allowed(vcpu))
    return 1;
@@ -2552,6 +2555,9 @@ static int vmx_set_msr(struct kvm_vcpu *vcpu, struct msr_data *msr_info)
  case MSR_IA32_SYSENTER_ESP:
   vmcs_writel(GUEST_SYSENTER_ESP, data);
   break;
+ case MSR_IA32_BNDCFGS:
+  vmcs_write64(GUEST_BNDCFGS, data);
+  break;
  case MSR_IA32_TSC:
   kvm_write_tsc(vcpu, msr_info);
   break;
diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 0bcf1a5..39f56cf 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -884,7 +884,7 @@ static u32 msrs_to_save[] = {
  MSR_CSTAR, MSR_KERNEL_GS_BASE, MSR_SYSCALL_MASK, MSR_LSTAR,
 #endif
  MSR_IA32_TSC, MSR_IA32_CR_PAT, MSR_VM_HSAVE_PA,
- MSR_IA32_FEATURE_CONTROL
+ MSR_IA32_FEATURE_CONTROL, MSR_IA32_BNDCFGS
 };
 
 static unsigned num_msrs_to_save;
-- 
1.7.1